service/src/main/java/org/apache/griffin/core/metastore/hive/HiveMetaStoreServiceJdbcImpl.java

```diff
@@ -0,0 +1,287 @@
+        } catch (ClassNotFoundException | SQLException e) {
+            e.printStackTrace();
+        }
+
+        try {
+            stmt = conn.createStatement();
+            rs = stmt.executeQuery(sql);
+            while (rs.next()) {
+                res.add(rs.getString(1));
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            closeConnection(conn, stmt, rs);
+        }
+        return res;
+    }
+
+    /**
+     * Get the Hive table location from hive table metadata string
+     * @param tableMetadata hive table metadata string
+     * @return Hive table location
+     */
+    public String getLocation(String tableMetadata) {
+        tableMetadata = tableMetadata.toLowerCase();
+        int index = tableMetadata.indexOf("location");
+        if (index == -1) {
+            return "";
+        }
+
+        int start = tableMetadata.indexOf("\'", index);
+        int end = tableMetadata.indexOf("\'", start + 1);
+
+        if (start == -1 || end == -1) {
+            return "";
+        }
+
+        return tableMetadata.substring(start + 1, end);
+    }
+
+    /**
+     * Get the Hive table schema: column name, column type, column comment
+     * @param tableMetadata hive table metadata string
+     * @return List of FieldSchema
+     */
+    public List<FieldSchema> getColums(String tableMetadata) {
+        List<FieldSchema> res = new ArrayList<>();
+        int start = tableMetadata.indexOf("(") + 1;
+        int end = tableMetadata.indexOf(")", start);
+        String[] colsArr = tableMetadata.substring(start, end).split(",");
+        for (String colStr : colsArr) {
+            colStr = colStr.trim();
+            String[] parts = colStr.split(" ");
+            String colName = parts[0].trim().substring(1, parts[0].trim().length() - 1);
+            String colType = parts[1].trim();
+            String comment = getComment(colStr);
+            FieldSchema schema = new FieldSchema(colName, colType, comment);
+            res.add(schema);
+        }
+        return res;
+    }
+
+    /**
+     * Parse one column string, such as : `merch_date` string COMMENT 'this is merch process date'
+     *
+     * @param colStr column string
+     * @return
+     */
+    public String getComment(String colStr) {
+        colStr = colStr.toLowerCase();
+        int i = colStr.indexOf("comment");
+        if (i == -1) {
+            return "";
+        }
+
+        int s = -1;
+        int e = -1;
+        while (i < colStr.length()) {
+            if (colStr.charAt(i) == '\'') {
+                if (s == -1) {
+                    s = i;
+                } else {
+                    e = i;
+                    break;
+                }
+            }
+            i++;
+        }
+        if (s == -1 || e == -1) {
+            return "";
+        }
+        if (s > e) {
+            return "";
+        }
+
+        return colStr.substring(s + 1, e);
+    }
+}
```

Comment:
Can we use regex to match ' ' and get column comment?  This seems not easy to understand.

Reply:
done